### ISSUE TITLE
Add section explaining what the github actions are

### ISSUE THREAD

**Issue by pdmthorsrud** [fredrikhgrelland/vagrant-hashistack#210]
When cloning the template repo you end up with automated git hub actions testing your repo, but this is not explained anywhere.

The tests work by doing mostly the same as this repo's tests, by spinning up a box and trying to run playbooks inside it.

**Comment by fredrikhgrelland** [comment]
Maybe those tests should be different and more limited?

**Comment by pdmthorsrud** [comment]
> Maybe those tests should be different and more limited?

In what dimension would you like to limit them? In amount of test scenarios in the matrix, or something else?

**Comment by fredrikhgrelland** [comment]
Yes, but also make them nudge you into writing meaningful tests for YOUR service, rather than using boilerplate

**Comment by pdmthorsrud** [comment]
> Yes, but also make them nudge you into writing meaningful tests for YOUR service, rather than using boilerplate

Right, so a very barebones framework that needs adding to then? Do we want to nudge them in the direction of adding their own github-actions? If we do wew might end up with some repos with amazing test-coverage, and some with basically none. By supplying one generic framework we make sure everyone at least has a basic amount of testing.

You're maybe talking about using the latter, and making it minimal?

**Comment by Neha-Sinha2305** [comment]
Can someone help me to scope out this issue?

Tasks:
1. Reduce the number of tests that are currently run as per the matrix testing strategy (run matrix testing for the default values : nomad_acl=false, consul_acl=true, consul_policy=deny)
2. ???????
3. ??????
@fredrikhgrelland @pdmthorsrud @zhenik

**Comment by zhenik** [comment]
I would also describe 3 possible options:
1. run all tests in CI (env variable CI)
2. run all tests in local (extra-vars --tags=local_test=true)
3. run without tests, when do development. (template/Makefile -> make dev)

**Comment by pdmthorsrud** [comment]
The comments above from Fredrik are not part of this issue. What I want is a section that details the github actions (what they are, and how to use them) that will follow when you clone the template.

**Comment by Neha-Sinha2305** [comment]
> The comments above from Fredrik are not part of this issue. What I want is a section that details the github actions (what they are, and how to use them) that will follow when you clone the template.

Ok..Is it good enough if we copy paste this section [test-configuration-and-execution](https://github.com/fredrikhgrelland/vagrant-hashistack#test-configuration-and-execution) from vagrant-hashistack/README.md to template/README.md as we aren't changing the tests for the template now?

**Comment by pdmthorsrud** [comment]
> > The comments above from Fredrik are not part of this issue. What I want is a section that details the github actions (what they are, and how to use them) that will follow when you clone the template.
> 
> Ok..Is it good enough if we copy paste this section [test-configuration-and-execution](https://github.com/fredrikhgrelland/vagrant-hashistack#test-configuration-and-execution) from vagrant-hashistack/README.md to template/README.md as we aren't changing the tests for the template now?

Good enough for the first iteration, but I would just a quick line explaining what is run during tests, so that people can potentially customise if they would like. :)
